test/libdnf/base/test_goal.cpp

```diff
@@ -52,20 +53,55 @@ void BaseGoalTest::test_install() {
 }
 
 void BaseGoalTest::test_install_from_cmdline() {
+    // this test covers an old dnf bug described in the following steps:
+    // * specify a commandline package to install
+    // * the specified package has the same NEVRA as an available package in a repo
+    // * the package query uses NEVRA instead of id and is resolved into the available rather than the specified package
+    // * -> an unexpected package is installed
+
+    // add a repo with package 'one-0:1-1.noarch'
+    add_repo_rpm("rpm-repo1");
+
+    // add 'one-0:1-1.noarch' package from the command-line
+    std::filesystem::path rpm_path = PROJECT_BINARY_DIR "/test/data/repos-rpm/rpm-repo1/noarch/one-1-1.noarch.rpm";
+    auto cmdline_pkg = sack->add_cmdline_package(rpm_path, false);
+
+    // check if the available package is there and has lower id than the command-line package
+    // a package with lower id would be normally preferred if the packages are identical
+    libdnf::rpm::SolvQuery query(&(base->get_rpm_solv_sack()));
+    query.ifilter_available().ifilter_nevra(libdnf::sack::QueryCmp::EQ, {"one-0:1-1.noarch"});
+
+    std::vector<libdnf::rpm::Package> query_packages;
+    for (auto i : query) {
+        query_packages.push_back(i);
+    }
+    // confirm that the second package is the command-line package
+    CPPUNIT_ASSERT_EQUAL(cmdline_pkg.get_id().id, query_packages[1].get_id().id);
+
+    // the available package has lower id than the command-line package
+    CPPUNIT_ASSERT(query_packages[0].get_id().id < query_packages[1].get_id().id);
```

Comment:
I would prefer to replace from line 69 to 82 by different test - availability of another package with the same NEVRA.
```
    // check whether there is other available package with the same NEVRA
    libdnf::rpm::SolvQuery query(&(base->get_rpm_solv_sack()));
    query.ifilter_available().ifilter_nevra(libdnf::sack::QueryCmp::EQ, {"one-0:1-1.noarch"});
    query.ifilter_repoid(libdnf::sack::QueryCmp::NEQ, {"@commandline"});

    // confirm the availability of "one-0:1-1.noarch" from available repo
    CPPUNIT_ASSERT_EQUAL(query.size(), 1);
```